src/Databases/TablesDependencyGraph.cpp

```diff
@@ -699,14 +699,14 @@ std::vector<StorageID> TablesDependencyGraph::getTablesSortedByDependency() cons
 }
 
 
-std::map<size_t, std::vector<StorageID>> TablesDependencyGraph::getTablesSortedByDependencyWithLevels() const
+std::vector<std::vector<StorageID>> TablesDependencyGraph::getTablesSplitByDependencyLevel() const
 {
-    std::map<size_t, std::vector<StorageID>> tables_by_level;
+    std::vector<std::vector<StorageID>> tables_split_by_level;
     for (const auto * node : getNodesSortedByLevel())
     {
-        tables_by_level[node->level].emplace_back(node->storage_id);
+        tables_split_by_level[node->level].emplace_back(node->storage_id);
```

Comment:
It will crash with segfault, you have to resize the vector first. Or use emplace_back/push_back